roles/pulp_health_check/tasks/main.yml

```diff
@@ -0,0 +1,35 @@
+---
+- meta: flush_handlers
+
+- name: Get services
+  service_facts:
+
+- name: Listing Pulp services
+  debug:
+    msg: '{{ ansible_facts.services|dict2items|selectattr("key", "contains", "pulp")|list|items2dict }}'
+
+
+- name: Checking Pulp services
+  fail:
+    msg: "{{ item.value.name }} state: {{ item.value.state }}"
+  when: "item.value.name != 'pulpcore-worker@.service' and item.value.state != 'running'"
+  with_dict: '{{ ansible_facts.services|dict2items|selectattr("key", "contains", "pulp")|list|items2dict }}'
+
+- name: Ensure Pulp is up and healthy
+  uri:
+    url: "http://{{ pulp_api_bind }}/pulp/api/v3/status/"
+    user: admin
+    password: "{{ pulp_default_admin_password }}"
+    force_basic_auth: yes
+    status_code: "200"
+    validate_certs: "{{ pulp_validate_certs }}"
```

Comment:
I might have mentioned this previously. Since we are hardcoding "http" (and intending to run against pulp_api directly, not the webserver), there's no point in validating certs.

Reply:
@mikedep333 it came from tower: https://github.com/chouseknecht/ansible_galaxy_config/pull/7

Reply:
If the pulp server is deployed with self-signed SSL then it will redirect anything from `http` to `https` and therefore it would be good to have the option to select if I want to validate the cert or not.

Reply:
@fao89 We do not have that variable on our master branch. And it is named differently.

@elyezer I thought that redirecting was done by the webserver, not pulp-api. Either way, the webserver isn't up yet when this role is run. And using `pulp_api_bind` to define the host doesn't make sense.

IMHO, we should have a separate variable like `pulp_health_check_origin` that defaults to `http://{{ pulp_api_bind }}`. Then `pulp_health_check_origin` is a role variable, and `pulp_api_bind` is a shared variable.

Reply:
@mikedep333 you are right, at galaxy role it is ran for the webserver and here we are running for the api

Reply:
Thanks for the extra context, I don't know much about the internals here. But I rely 100% on everything you're saying :smile: